compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -169,7 +169,16 @@ impl<'a, 'tcx> Encodable<EncodeContext<'a, 'tcx>> for ExpnId {
 impl<'a, 'tcx> Encodable<EncodeContext<'a, 'tcx>> for Span {
     fn encode(&self, s: &mut EncodeContext<'a, 'tcx>) {
         match s.span_shorthands.entry(*self) {
-            Entry::Occupied(o) => SpanEncodingMode::Shorthand(*o.get()).encode(s),
+            Entry::Occupied(o) => {
+                // If an offset is smaller than the absolute position, we encode with the offset.
+                // This saves space since smaller numbers encode in less bits.
+                let offset = s.opaque.position() - *o.get();
```

Comment:
since `*o.get()` is used multiple times, it would be nicer to have a variable for it like `let &addr = o.get();`.

Reply:
Can do.